csharp-intermediate-topics/RenameFilesWithCSharp/Tests/RenameFilesUnitTests.cs

```diff
@@ -0,0 +1,132 @@
+        Directory.CreateDirectory(tempDirectory);
+
+        File.Create(Path.Combine(tempDirectory, "photo1.jpg")).Close();
+        File.Create(Path.Combine(tempDirectory, "photo2.png")).Close();
+
+        // Act
+        BatchRenameFiles.RenamePhotosWithDateTime(tempDirectory);
+
+        // Assert
+        Assert.True(File.Exists(Path.Combine(tempDirectory, $"Image_{DateTime.Now:yyyy-MM-dd_HHmmss}.jpg")));
+        Assert.True(File.Exists(Path.Combine(tempDirectory, $"Image_{DateTime.Now:yyyy-MM-dd_HHmmss}.png")));
+
+        // Clean up
+        Directory.Delete(tempDirectory, true);
+    }
+
+
+    [Fact]
+    public void WhenBatchRenamingPhotosWithDateTimeAndNoAllowedExtensionsExist_ThenNoPhotosShouldBeRenamed()
+    {
+        // Arrange
+        var tempDirectory = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+        Directory.CreateDirectory(tempDirectory);
+
+        File.Create(Path.Combine(tempDirectory, "document.docx")).Close();
+
+        // Act
+        BatchRenameFiles.RenamePhotosWithDateTime(tempDirectory);
+
+        // Assert
+        Assert.False(File.Exists(Path.Combine(tempDirectory, $"Image_{DateTime.Now:yyyy-MM-dd_HHmmss}.jpg")));
+        Assert.False(File.Exists(Path.Combine(tempDirectory, $"Image_{DateTime.Now:yyyy-MM-dd_HHmmss}.png")));
+
+        // Clean up
+        Directory.Delete(tempDirectory, true);
+    }
+
+
+    [Fact]
+    public void WhenRenamingPhotosWithDateTimeAllowedExtensionsExist_ThenPhotosShouldBeRenamed()
+    {
+        // Arrange
+        var tempDirectory = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+        Directory.CreateDirectory(tempDirectory);
+
+        File.Create(Path.Combine(tempDirectory, "photo1.jpg")).Close();
+        File.Create(Path.Combine(tempDirectory, "photo2.png")).Close();
+
+        // Act
+        BatchRenameFileWithErrorHandling.RenamePhotosWithDateTime(tempDirectory);
+
+        // Assert
+        Assert.True(File.Exists(Path.Combine(tempDirectory, $"Image_{DateTime.Now:yyyy-MM-dd_HHmmss}.jpg")));
+        Assert.True(File.Exists(Path.Combine(tempDirectory, $"Image_{DateTime.Now:yyyy-MM-dd_HHmmss}.png")));
+
+        // Clean up
+        Directory.Delete(tempDirectory, true);
+    }
+
+    [Fact]
+    public void WhenRenamingPhotosWithDateTimeAndNoAllowedExtensionsExist_ThenNoPhotosShouldBeRenamed()
+    {
+        // Arrange
+        var tempDirectory = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+        Directory.CreateDirectory(tempDirectory);
+
+        File.Create(Path.Combine(tempDirectory, "document.docx")).Close();
+
+        // Act
+        BatchRenameFileWithErrorHandling.RenamePhotosWithDateTime(tempDirectory);
+
+        // Assert
+        Assert.False(File.Exists(Path.Combine(tempDirectory, $"Image_{DateTime.Now:yyyy-MM-dd_HHmmss}.jpg")));
+        Assert.False(File.Exists(Path.Combine(tempDirectory, $"Image_{DateTime.Now:yyyy-MM-dd_HHmmss}.png")));
+
```

Comment:
See earlier comment on similar test, the assertion here should be that no file was renamed and so we should get the list of files, run the rename method, then assert that the files all have the same name as before.